module7/Homework7-start/Homework7Tests/UserStoreTests.swift

```diff
@@ -0,0 +1,113 @@
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+/// THE SOFTWARE.
+
+import XCTest
+
+final class UserStoreTests: XCTestCase {
+  var userStore: UserStore!
+  let missingURL = URL(fileURLWithPath: "missingFile",
+                       relativeTo: FileManager.documentsDirectoryURL).appendingPathExtension("json")
+  
+  
+  override func setUpWithError() throws {
+    userStore = UserStore()
+  }
+  
+  override func tearDownWithError() throws {
+    if FileManager.default.fileExists(atPath: userStore.documentsJSONURL.path) {
+      try FileManager.default.removeItem(at: userStore.documentsJSONURL)
+    }
+  }
+  
+  func testAppStoreInitialState() throws {
+    XCTAssertNotNil(userStore, "userStore should initialize but was nil")
+    XCTAssertNil(userStore.userData, "Initial userStore.userData should be nil")
+    XCTAssertEqual(userStore.dataState, .notLoaded, "Intial userStore.dataState should be .notLoaded")
+  }
+  
+  func testReadAPIJSONFromUrl() throws {
+    let apiData = userStore.readJSONFromUrl(url: userStore.bundleJSONURL)
+    XCTAssertNotNil(apiData, "userStore.readAPIJSONFromUrl(url: userStore.bundleJSONURL) should read non-nil data")
+  }
+
+  func testMissingUrl() throws {
+    let fileExistsAtMissingPath = FileManager.default.fileExists(atPath: missingURL.path)
+    XCTAssertEqual(fileExistsAtMissingPath, false, "No file should exist at missingURL.path")
+  }
+  
+  func testReadAPIJSONFromPrimaryUrl() throws {
+    let userData = try userStore.readJSON(with: userStore.bundleJSONURL, fallingBackTo: missingURL)
+    XCTAssertNotNil(userData, "userStore.readAPIJSON(with:fallingBackTo:) successfully reads from primary url")
+  }
+
+  func testReadAPIJSONFromFallbackUrl() throws {
+    let userData = try userStore.readJSON(with: missingURL, fallingBackTo: userStore.bundleJSONURL)
+    XCTAssertNotNil(userData, "userStore.readJSON(with:fallingBackTo:) successfully falls back to secondary url when primary url does not exist")
+  }
+
+  func testReadUserJSON() throws {
+    userStore.readJSON()
+    XCTAssertNotNil(userStore.data,
+                   "After successful load userStore.data should not be empty")
+    XCTAssertEqual(userStore.dataState, .loaded,
+                   "After successful load, userStore.dataState should be .loaded")
+  }
+
+  func testReadJSONFromFallbackUrlThrowsError() {
+    XCTAssertThrowsError(
+      try userStore.readJSON(with: missingURL, fallingBackTo: missingURL),
+      "", { error in
+        let error = error as! JSONDataLoadingStoreError
+        XCTAssertEqual(
+          error,
+          JSONDataLoadingStoreError.DataFileNotFound("Api data file not found at primaryURL: \(missingURL) or \(missingURL)"),
+          "userStore.readJSON(with:fallingBackTo:) should throw JSONDataLoadingStoreError.DataFileNotFound on missing paths")
```

Comment:
You've already got a lot going on here, but I often add to my error messages in tests what value I _actually_ got back. It helps speed up the troubleshooting.